packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -90,7 +90,7 @@ export const cleanTrailingZeroInNumericSectionValue = <TDate>(
 
 export const getLetterEditingOptions = <TDate>(
   utils: MuiPickersAdapter<TDate>,
-  dateSectionName: MuiDateSectionName,
+  dateSectionName: FieldSectionType,
```

Comment:
Would probably make sense to rename `dateSectionName` here as well.
Maybe go with `dateSectionType`, or `sectionType`, or `fieldType`, or `sectionFIeldType`. 🤔